frame/contracts/src/exec.rs

```diff
@@ -510,9 +502,35 @@ where
 		Ok((dest, output))
 	}
 
-	fn new_call_context<'b>(&'b mut self, caller: T::AccountId, value: BalanceOf<T>)
-		-> CallContext<'b, 'a, T, V, L>
-	{
+	pub fn terminate(
+		&mut self,
+		beneficiary: &T::AccountId,
+		gas_meter: &mut GasMeter<T>,
+	) -> Result<(), DispatchError> {
+		let self_id = self.self_account.clone();
```

Comment:
Can we get away without this clone?

Reply:
The problem here is that the `transfer` call borrows `self` mutably with the `ctx` arg. Therefore we cannot borrow `self.self_account` for the `transactor` arg.

In order to safe the clone we need to change how this function words.